### ISSUE TITLE
@decorateMethod

### ISSUE THREAD

**Issue by sonhanguyen** [jayphelps/core-decorators#143]
wonder what do you think about adding a `@decorateMethod` to convert method decorators to class decorators? The usecase is implementing the actual decorator pattern. Which using method decorators would look like this:
```ts
class ADecorator extends A { // unfortunately Proxy can't be base class
    @log
    @cache
    a(...args) {
        return super.a(...args)
    }

    @log
    b(...args) {
        return super.b(...args)
    }
}
```
which does look a bit repetitive, I'm proposing the following 
```ts
@decorateMethod({
   a: [cache, log] // order
   b: log
})
class ADecorator extends A {}

// since ADecorator is empty in this case, it may as well be simplified:
const ADecorator = decorateMethod({
   a: [cache, log]
   b: log
})(A)
```

looking for some hacktoberfest contribution so just checking if I should open a pull request

**Comment by jayphelps** [comment]
Shoot, I'm not following 😞  Could you maybe reword with some more real-world examples?

**Comment by sonhanguyen** [comment]
sorry reasoning being not fully fleshed, main motivation is that I'm not fan of method decorators. The decorator pattern like many other OOP patterns follow the open/closed principle: geven a thing, without modifying the thing source code, enhance it.

Method decorators require you to do this source code modification, which is sometimes possible sometimes isn't. e.g. given a 3rd library
```
export default AbcClient {
   doThing() {
   ...
}
```
now if I want to apply the decorator `@time` to `doThing`. Of course I can't just go into `node_modules` and annotate the method. I'll instead have to do the following:
```
import AbcClient as OriginalAbcClient from 'abc'
class AbcClient extends OriginalAbcClient {
   @time
   doThing() {
       super.doThing(arguments)
   }
   ...
}
\\ rest of the code unchanged 
```
which is verbose as said, with this hypothetical `decorateMethod`, I can instead do
```
const AbcClient = decorateMethod({doThing: time})
    (OriginalAbcClient)
```

**Comment by sonhanguyen** [comment]
actually nvm, I just saw we have `applyDecorators` which is the exact thing
